src/sidebar/components/test/login-control-test.js

```diff
@@ -1,115 +1,284 @@
 'use strict';
 
 var angular = require('angular');
+var proxyquire = require('proxyquire');
 
+var bridgeEvents = require('../../../shared/bridge-events');
 var util = require('../../directive/test/util');
 
-function PageObject(element) {
-  this.unknownLoginText = function () {
-    return element[0].querySelector('.login-text').textContent;
-  };
-  this.loginText = function () {
-    return Array.from(element[0].querySelectorAll('.login-text a'))
-      .map(function (el) { return el.textContent; });
-  };
-  this.menuLinks = function () {
-    return Array.from(element[0].querySelectorAll('.login-control-menu .dropdown-menu a'))
-      .map(function (el) { return el.textContent; });
-  };
-  this.disabledMenuItems = function () {
-    return Array.from(element[0].querySelectorAll('.login-control-menu .dropdown-menu__link--disabled'))
-      .map(function (el) { return el.textContent.trim(); });
-  };
-  this.menuText = function () {
-    return element[0].querySelector('span').textContent;
+function pageObject(element) {
+  return {
+    menuLinks: function () {
+      return Array.from(element[0].querySelectorAll('.login-control-menu .dropdown-menu a'))
+                  .map(function (el) { return el.textContent; });
+    },
+    menuText: function () {
+      return element[0].querySelector('span').textContent;
+    },
+    userProfileButton: element[0].querySelector('.dropdown-menu__user-profile-btn'),
```

Comment:
I'm not sure if we do this elsewhere or not, but if we have classes that exist purely to help locate elements of a DOM tree in tests I think it would be better to use `js-` names (eg. `js-user-profile-btn`) to make it obvious that these classes are purely aids for locating the element and not styled in any way.